File: main/java/com/laboManager/service/EquipmentService.java
package com.laboManager.service;

import com.laboManager.entity.PageResult;
import com.laboManager.entity.Result;
import com.laboManager.pojo.TbEquipment;
import com.laboManager.pojo.TbRepair;
import com.laboManager.pojo.TbScrap;

public interface EquipmentService {
	
	/**
	 * 查询设备列表
	 * @param pages
	 * @param rows
	 * @param equipment
	 * @return
	 */
	public PageResult search(int pages,int rows,TbEquipment equipment);

	/**
	 * 添加设备
	 * @param equipment
	 */
	public void add(TbEquipment equipment);
	
	/**
	 * 根据id查设备
	 * @param id
	 * @return
	 */
	public TbEquipment findEquipmentById(int id);
	
	/**
	 * 修改
	 * @param equipment
	 */
	public void update(TbEquipment equipment);
	
	/**
	 * 维修,报修设备
	 */
	public void repair(TbRepair repair);

	/**
	 * 报废
	 * @param scrap
	 */
	public void scrap(TbScrap scrap);
}

File: main/webapp/js/service/equipmentService.js
app.service("equipmentService",function($http){
	//查询设备列表
	this.search = function(pages,rows,equipment){
		return $http.post("equipment/search.do?pages="+pages+"&rows="+rows,equipment);
	}
	
	//保存
	this.save = function(methodName,equipment){
		return $http.post("equipment/"+methodName+".do",equipment);
	}
	
	//根据id查设备
	this.findEquipmentById = function(id){
		return $http.get("equipment/findEquipmentById.do?id="+id);
	}
	
	//维修
	this.repair = function(repair){
		return $http.post("equipment/repair.do",repair);
	}
	
	//报废
	this.scrap = function(scrap){
		return $http.post("equipment/scrap.do",scrap);
	}
}) 